src/preamble.js

```diff
@@ -130,14 +130,23 @@ function ccall(ident, returnType, argTypes, args, opts) {
     });
   }
 #else // EMTERPRETIFY_ASYNC
+  if (typeof Asyncify === 'object' && Asyncify.currData !== null) {
 #if ASSERTIONS
-  assert(!(opts && opts.async), 'async call is only supported with Emterpretify for now, see #9029');
+    assert(opts && opts.async, 'The call to ' + ident + ' is running asynchronously. If this was intended, add the async option to the ccall/cwrap call.');
+    assert(Asyncify.asyncFinalizers.length === 0, 'Cannot have multiple async ccalls in flight at once');
```

Comment:
If the assertion for `Asyncify.asyncFinalizers.length === 0` fails, that means that another `async ccall` is currently in-flight and waiting to have it's finalizers called, since once the finalizers are called, `asyncFinalizers` gets set back to `[]`.

Reply:
I think this an other comments in the PR could be comments in the source. Then they won't be lost later.